File: ch19p655saveAndRestoreArbitraryObject/ch19p655saveAndRestoreArbitraryObject/Thing.m


#import "Thing.h"

@interface Thing ()
@end

@implementation Thing

-(void)encodeRestorableStateWithCoder:(NSCoder *)coder {
    NSLog(@"%@", @"thing encode");
    [coder encodeObject:self.word forKey:@"word"];
}

-(void)decodeRestorableStateWithCoder:(NSCoder *)coder {
    NSLog(@"%@", @"thing decode");
    self.word = [coder decodeObjectForKey:@"word"];
}

-(void)applicationFinishedRestoringState {
    NSLog(@"%@", @"finished thing");
}

@end

File: iOS9bookExamples/bk2ch06p356saveAndRestoreArbitraryObject/ch19p655saveAndRestoreArbitraryObject/Thing.swift


import UIKit

class Thing : NSObject, UIStateRestoring {
    
    var word = ""
    
    var restorationParent: UIStateRestoring? // unused
    
    var objectRestorationClass: AnyObject.Type? // unused
    
    func encodeRestorableStateWithCoder(coder: NSCoder) {
        print("thing encode")
        coder.encodeObject(self.word, forKey:"word")
    }
    
    func decodeRestorableStateWithCoder(coder: NSCoder) {
        print("thing decode")
        self.word = coder.decodeObjectForKey("word") as! String
    }
    
    func applicationFinishedRestoringState() {
        print("finished thing")
    }
    
    
}
